helix-term/src/commands.rs

```diff
@@ -2439,6 +2600,29 @@ fn replace_with_yanked(cx: &mut Context) {
     }
 }
 
+fn replace_selections_with_clipboard_impl(editor: &mut Editor) {
+    let (view, doc) = current!(editor);
+
+    match editor.clipboard_provider.get_contents() {
+        Ok(contents) => {
+            let transaction =
+                Transaction::change_by_selection(doc.text(), doc.selection(view.id), |range| {
+                    let max_to = doc.text().len_chars().saturating_sub(1);
+                    let to = std::cmp::min(max_to, range.to() + 1);
```

Comment:
This pattern is now frequent enough that maybe we should add a helper function `doc.clipped_range` or something, it would help with #224. Can be done over there I guess

Reply:
If it helps in this PR, maybe it should be done there indeed. I'll make sure to use the helper if it gets merged before my PR.